### PR TITLE
Add $APPEND_IF_NO_NEWLINE environment variable

### LINKED ISSUE #448: Prompt displayed on the same line as command output
Consider the following code:

```
echo -n foo
```

. This will result in

```
foozyx@zyx-desktop ~ $
```

on the next line in place of just

```
zyx@zyx-desktop ~ $
```

. Zsh in this case outputs per cent sign:

highlighted per cent after foo

(note: attaching a screenshot to show that this sign is highlighted). (Of course, in zsh this is controlled by some option. I do not see why this should be optional though, but in zsh you can customize not only sign presence, but also how it is highlighted and what text should be there exactly (including running some command and displaying its output).)

Xonsh version 0.2.2.

### PR DESCRIPTION
**Pull request by rossnomann**
Closes #448

### CODE REVIEW

**Comment by laerus** [comment]
Thanks for adding this feature!!

A better name would be `XONSH_APPEND_NEWLINE` or something like this but with the `XONSH_` prefix.

**Comment by scopatz** [comment]
Thanks for putting this in @rossnomann! Please let us know when it is ready for another look!

**Comment by rossnomann** [comment]
@scopatz @laerus I've updated PR.

**Comment by scopatz** [comment]
Hi @rossnomann - thanks for putting this in! I tend agree with @laerus, though, that `$XONSH_APPEND_NEWLINE` is a better name.  Any reason not to change it?

**Comment by rossnomann** [comment]
Hi @scopatz  Well, I'll rename it - I don't see the difference.

**Comment by scopatz** [comment]
Adding XONSH means that it shouldn't collide with other variables in the environment.

**Comment by rossnomann** [comment]
@scopatz I know. I meant APPEND_**IF_NO**_NEWLINE vs APPEND_NEWLINE.

**Comment by scopatz** [comment]
Oh, it is generally a good idea to keep flag names positive, since some people find them easier to reason about.

**Comment by rossnomann** [comment]
Good point :+1: . I already heard about it, but cant remember. Thank you!

File: xonsh/base_shell.py [hunk 1]
```diff
@@ -339,8 +339,12 @@ def default(self, line):
                 hist.last_cmd_rtn = 1  # return code for failure
         finally:
             ts1 = ts1 or time.time()
-            self._append_history(inp=src, ts=[ts0, ts1], tee_out=tee.getvalue())
+            tee_out = tee.getvalue()
+            self._append_history(inp=src, ts=[ts0, ts1], tee_out=tee_out)
             self.accumulated_inputs += src
+            append_newline = env.get('XONSH_APPEND_NEWLINE')
+            if append_newline and not tee_out.endswith(os.linesep):
+                print(append_newline, end='')
```
Comment (laerus) [inline xonsh/base_shell.py]: isn't this suppose to print a new line and not the envvar?

Comment (laerus) [inline xonsh/base_shell.py]: if you want to print arbitary text after a command you can always use the on_post_cmd event.

Comment (rossnomann) [inline xonsh/base_shell.py]: > isn't this suppose to print a new line and not the envvar?
We need to print the envvar, of course.
> you can always use the on_post_cmd event
Can I get previous output and check for os.linesep at the end in this case?

Comment (rossnomann) [inline xonsh/base_shell.py]: Can't find `on_post_cmd` event.

Comment (laerus) [inline xonsh/base_shell.py]: sorry it is `events.on_postcommand`, yeah you can check various attributes of the last command.
try:
`print(events.on_postcommand.__doc__)`

i think we should just print a newline and leave more complicated cases for the events

Comment (rossnomann) [inline xonsh/base_shell.py]: Ok. So, I should print `os.linesep` right here, right?

**Comment by rossnomann** [comment]
@laerus I've updated PR

**Comment by laerus** [comment]
thanks @rossnomann !!!

**Pull request merged**